api/client/run.go

```diff
@@ -267,28 +258,22 @@ func (cli *DockerCli) CmdRun(args ...string) error {
 	var status int
 
 	// Attached mode
-	if *flAutoRemove {
-		// Autoremove: wait for the container to finish, retrieve
-		// the exit code and remove the container
+	if !config.Tty && hostConfig.RestartPolicy.IsNone() {
```

Comment:
If we're going to allow to have `--rm` with `--restart` I think the client should wait forever regardless.

Allowing restarting and removal together may actually be a bit too much and confusing from a client point of view. What scenarios is this combination supposed to cater for?